### ISSUE TITLE
Two entries with the same name in different directories lead to "Two output files share the same path"

### ISSUE THREAD

**Issue by matthewmueller** [evanw/esbuild#244]
**Problem**

Given the following file structure:

```
.
├── a
│   └── index.js
└── b
    └── index.js
```

If you run:

```
esbuild --bundle --outdir=dist --format=esm a/index.js b/index.js
```

You get the following error:

```
error: Two output files share the same path: dist/index.js
```

**Possible Solution**

I think I'd expect the bundle output to look like this:

```
dist
├── a
│   └── index.js
└── b
    └── index.js
```

So it seems like there'd need to be a base directory (defaulting to working directory) so you could do:

```js
const rel = path.relative(basedir, entry)
const outpath = path.join(outdir, rel)
```

**Comment by nitsky** [comment]
@matthewmueller see #224 which reports the same issue and suggests a solution similar to how Rollup handles this.

**Comment by matthewmueller** [comment]
Thanks, I see it now. I personally think this issue is a bit more focused on the problem. Having the exact error message in the issue title makes it more discoverable for others too.

I agree that customizing the names of chunks, assets, and entrypoints is nice to have (I use these features in rollup too), but even without any of those options, rollup and tsc handle this for you.

Let's do this. I'll close this issue and write a comment in #224.

**Comment by evanw** [comment]
This should be fixed as of version 0.6.0.
